src/greeter/GreeterApp.h

```diff
@@ -38,7 +38,7 @@ namespace SDDM {
     class KeyboardModel;
 
 
-    class GreeterApp : public QGuiApplication
+    class GreeterApp : public QApplication
```

Comment:
Simply delaying the closing of the greeter on the greeter's side (until all PAM messages are confirmed) works not as expected and causes strange looking behaviour like fade-in effects which will also affect the greeter when the user's session is started in the background (already tried that).

So the way to go would be to tell the helper to start the user's session only when all PAM messages are confirmed, which requires additional communication from the greeter to the helper, like lightDM does.

I don't think the Admin-Info box is suited that well, as these are two different components and we would have to implement a confirmation-button and reset-functionality for only one of them (and remove it for the other) and disable dynamic updates while PAM-messages are displayed to not overwrite the PAM-messages, which will affect readability of the code and I'm not sure whether the user will notice it as good this way (and not be confused as the Admin-Info box contains important information as well).